### PR TITLE
Promote StatefulSet Replica scaling

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promotes the following E2E test to Conformance:
test/e2e/apps/statefulset.go: "should have a working scale subresource"

**Special notes for your reviewer**:
Part of Umbrella Issue #78747.
[Testgrid results](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&include-filter-by-regex=should%20surface%20a%20failure%20condition%20on%20a%20common%20issue%20like%20exceeded%20quota&include-filter-by-regex=ReplicaSet%20should%20surface%20a%20failure%20condition%20on%20a%20common%20issue%20like%20exceeded%20quota&width=5&include-filter-by-regex=%5Bsig-apps%5D%20StatefulSet%20%5Bk8s.io%5D%20Basic%20StatefulSet%20functionality%20%5BStatefulSetBasic%5D%20should%20have%20a%20working%20scale%20subresource).
Test(s):
- [sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should have a working scale subresource  

Endpoint(s):
- replaceAppsV1NamespacedStatefulSetScale
- readAppsV1NamespacedStatefulSetScale

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```
/area conformance
/area test
@kubernetes/sig-apps-pr-reviews
@kubernetes/sig-architecture-pr-reviews
@kubernetes/cncf-conformance-wg

### CODE REVIEW

File: test/e2e/apps/statefulset.go [hunk 1]
```diff
@@ -762,7 +762,14 @@ var _ = SIGDescribe("StatefulSet", func() {
 			}, e2esset.StatefulPodTimeout, 2*time.Second).Should(gomega.BeNil())
 		})
 
-		ginkgo.It("should have a working scale subresource", func() {
+		/*
+						Release : v1.16
```
Comment (draveness) [inline test/e2e/apps/statefulset.go]: Please reformat this comment.

File: test/e2e/apps/statefulset.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega"
 	appsv1 "k8s.io/api/apps/v1"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (draveness) [inline test/e2e/apps/statefulset.go]: Looks good to me, but other reviewers would want this to be removed.

Comment (BobyMCbobs) [reply test/e2e/apps/statefulset.go]: Undid that. Appears to have change from my linter.

File: test/e2e/apps/statefulset.go (hunk 1 above)
Comment (BobyMCbobs) [reply test/e2e/apps/statefulset.go]: I've fixed the formatting, should be better now :smile:

**Comment by spiffxp** [comment]
/assign @bgrant0607 @smarterclayton 
for the /approve

**Pull request merged**